spec.emu

```diff
@@ -882,6 +882,297 @@ contributors: Google, Ecma International
+          1. Set _monthsInYear_ to CalendarMonthsInYear(_calendar_, _resolvedYear_).
+          1. Set _resolvedMonth_ to _resolvedMonth_ + _monthsInYear_.
+        1. Repeat, while _resolvedMonth_ &gt; _monthsInYear_,
+          1. Set _resolvedMonth_ to _resolvedMonth_ - _monthsInYear_.
+          1. Set _resolvedYear_ to _resolvedYear_ + 1.
+          1. Set _monthsInYear_ to CalendarMonthsInYear(_calendar_, _resolvedYear_).
+        1. Let _resolvedDay_ be _day_.
+        1. Let _daysInMonth_ be CalendarDaysInMonth(_calendar_, _resolvedYear_, _resolvedMonth_).
+        1. Repeat, while _resolvedDay_ &le; 0,
+          1. Set _resolvedMonth_ to _resolvedMonth_ - 1.
+          1. If _resolvedMonth_ is 0, then
+            1. Set _resolvedYear_ to _resolvedYear_ - 1.
+            1. Set _monthsInYear_ to CalendarMonthsInYear(_calendar_, _resolvedYear_).
+            1. Set _resolvedMonth_ to _monthsInYear_.
+          1. Set _daysInMonth_ to CalendarDaysInMonth(_calendar_, _resolvedYear_, _resolvedMonth_).
+          1. Set _resolvedDay_ to _day_ + _daysInMonth_.
+        1. Repeat, while _resolvedDay_ &gt; _daysInMonth_,
+          1. Set _resolvedDay_ to _resolvedDay_ - _daysInMonth_.
+          1. Set _resolvedMonth_ to _resolvedMonth_ + 1.
+          1. If _resolvedMonth_ &gt; _monthsInYear_, then
+            1. Set _resolvedYear_ to _resolvedYear_ + 1.
+            1. Set _monthsInYear_ to CalendarMonthsInYear(_calendar_, _resolvedYear_).
+            1. Set _resolvedMonth_ to 1.
+          1. Set _daysInMonth_ to CalendarDaysInMonth(_calendar_, _resolvedYear_, _resolvedMonth_).
+        1. Return the Record { [[Year]]: _resolvedYear_, [[Month]]: _resolvedMonth_, [[Day]]: _resolvedDay_ }.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-nonisodatesurpasses" type="implementation-defined abstract operation">
+      <h1>
+        NonISODateSurpasses (
+          _calendar_: a calendar type that is not *"iso8601"*,
+          _sign_: -1 or 1,
+          _baseDate_: an ISO Date Record,
+          _isoDate2_: an ISO Date Record,
+          _years_: an integer,
+          _months_: an integer,
+          _weeks_: an integer,
+          _days_: an integer,
+        ): a Boolean
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>
+          The return value indicates whether the date _date1_, the result of adding the duration denoted by _years_, _months_, _weeks_, and _days_ to _baseDate_ in the calendar system denoted by _calendar_, surpasses _isoDate2_ in the direction denoted by _sign_.
+          If _weeks_ and _days_ are both zero, then _date1_ need not exist (for example, it could be February 30).
+        </dd>
+      </dl>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. Let _parts_ be CalendarISOToDate(_calendar_, _baseDate_).
+        1. Let _yearMonth_ be RegulateNonISOYearMonth(_calendar_, _parts_.[[Year]], _parts_.[[MonthCode]], _years_).
```

Comment:
thought related to previous: My immediate response to this was "shouldn't the result of `CalendarISOToDate` already be regulated" before I remembered that `RegulateNonISOYearMonth` also does some year addition.

Reply:
I acknowledge that this was confusing, and it should be fixed in 6abbe6e3ca1ca220e33f3a62cb474fe631a75495